bot/exts/fun/quack.py

```diff
@@ -0,0 +1,63 @@
+from typing import Optional
+
+import discord
+from aiohttp import client_exceptions
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours
+
+API_URL = 'https://quackstack.pythondiscord.com'
+
+
+class Quack(commands.Cog):
+    """Cog for the .quack command."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+    @commands.command()
+    async def quack(self, ctx: commands.Context, ducktype: Optional[str] = "duck", seed: Optional[str] = None) -> None:
+        """
+        Uses the Quackstack API to generate a random duck.
+
+        If a seed is provided, a duck is generated based on the given seed.
+        If a duck type is provided, either a duck or manduck will be generated.
+        """
+        ducktype = ducktype.lower()
+        if ducktype not in ["duck", "manduck"]:
+            raise commands.BadArgument(
+                "The duck type can only be `duck` or `manduck`."
+            )
+
+        quackstack_url = f"{API_URL}/{ducktype}"
+        params = {}
+        if seed is not None:
+            try:
+                seed = int(seed)
+            except ValueError:
+                # Converts any words or words with numbers into an integer using ord()
+                seed = int(''.join([str(ord(e)) for e in seed]))
+            params["seed"] = seed
+
+        try:
+            async with self.bot.http_session.get(quackstack_url, params=params) as r:
```

Comment:
Still doesn't handle non-200 responses, which was mostly what I thought we should handle. We don't catch `ClientConnectorError` elsewhere afaik.